### ISSUE TITLE
Importing fuzzywuzzy unleashed a flood of warnings 

### ISSUE THREAD

**Issue by samkennerly** [seatgeek/fuzzywuzzy#144]
## problem

After I import `fuzzywuzzy` into a script or notebook, I often see obscure warnings like this one:
```
/opt/conda/lib/python3.5/site-packages/gspread/client.py:252: ResourceWarning: unclosed <ssl.SSLSocket fd=49, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('123.45.67.89', 43696), raddr=('123.45.678.901', 443)>
  client.login()
```
In this example, I imported a custom package which imports `fuzzywuzzy`, but I didn't actually use `fuzzywuzzy` at all.

The stack trace points to the `gspread` package, which led me on a long wild goose chase studying how `requests` uses HTTP persistent connections. I tried wrapping statements in `warnings.catch_warnings()` to catch errors, but that didn't work. I even [disabled *all* Python warnings](http://stackoverflow.com/a/14463321) with `python -W ignore`. But the warnings just kept coming.

## explanation

I think the culprit is line 33 of [process.py](https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/process.py):
```
warnings.simplefilter('always')
```
By default, [Python suppresses certain warnings](https://docs.python.org/3/library/warnings.html#default-warning-filters). But `warnings.simplefilter('always')` switches those warnings back on! Then my notebooks and log files become polluted with inscrutable `ResourceWarning` and `DeprecationWarning` statements about complicated things I'd rather ignore.

Removing `import fuzzywuzzy` made the warnings go away. But I want my team to keep using `fuzzywuzzy` because IMO, it's a clever tool that's mercifully easy to use.

## workaround

For now, my custom package has this in `__init__.py` after `import fuzzywuzzy`:
```
# Hack to partially fix paranoid warnings from fuzzywuzzy
import warnings
warnings.filterwarnings('ignore','.*',ResourceWarning)
warnings.filterwarnings('ignore','.*',DeprecationWarning)
warnings.filterwarnings('ignore','.*',PendingDeprecationWarning)
```
This (roughly?) restores the usual rules for suppressing warnings.

## permanent fix?

I think it's enough to move `warnings.simplefilter()` inside a `with warnings.catch_warnings()` block like [this](https://docs.python.org/3/library/warnings.html#testing-warnings).

**Comment by josegonzalez** [comment]
I'm personally of the opinion that warnings are there for a reason, and should be fixed in your code (or upstream libraries where possible).

That said, I defer to @acslater00 as to the best way to handle this. Perhaps we can just change this to a `logger.warning` call instead?

**Comment by samkennerly** [comment]
The `fuzzywuzzy` warning itself is no problem at all!

But altering the Python warning suppression rules causes me to see multiple `ResourceWarning` statements every time I open a connection using `gspread`, even in scripts where I never call `fuzzywuzzy` and everything else is working OK. (Those warnings are caused by the [somewhat elaborate way `requests` persists connections](https://github.com/kennethreitz/requests/issues/1882).)

I sometimes also see multiple `PendingDeprecationWarning` warnings from `import matplotlib` if there's even the hint of a future dependency conflict with another Python library. It's nothing fatal... just very distracting for the 99% of scripts I write which don't use fuzzy string matching.

**Comment by josegonzalez** [comment]
Released as part of 0.14.0.
